langchain/document_loaders/url.py

```diff
@@ -94,7 +105,15 @@ def load(self) -> List[Document]:
                     continue
                 else:
                     raise e
-            text = "\n\n".join([str(el) for el in elements])
-            metadata = {"source": url}
-            docs.append(Document(page_content=text, metadata=metadata))
+
+            if self.mode == "single":
+                text = "\n\n".join([str(el) for el in elements])
+                metadata = {"source": url}
+                docs.append(Document(page_content=text, metadata=metadata))
+            elif self.mode == "elements":
+                for element in elements:
+                    metadata = element.metadata.to_dict()
```

Comment:
should we still include source? is that already part of this?

Reply:
@dev2049 - Checking in to see if you wanted me to change anything here.